### PR TITLE
[Automatic Hierarchy Management] Rename IHM to automatic hierarchy management and move out of beta header

### PR DESCRIPTION
**Pull request by maicki**
Will replace #2022 that was accidentally closed.

### CODE REVIEW

File: AsyncDisplayKit/ASDisplayNode.mm [hunk 1]
```diff
@@ -1874,7 +1864,7 @@ - (void)__setSupernode:(ASDisplayNode *)newSupernode
       
       // If a node was added to a supernode, the supernode could be in a layout pending state. All of the hierarchy state
       // properties related to the transition need to be copied over as well as propagated down the subtree.
-      // This is especially important as with Implicit Hierarchy Management adding subnodes can happen while a transition
+      // This is especially important as with automatic hierarchy management, adding subnodes can happen while a transition
```
Comment (Adlai-Holler) [inline AsyncDisplayKit/ASDisplayNode.mm]: Replace "hierarchy management" by "subnode management" here and everywhere

**Comment by maicki** [comment]
@Adlai-Holler Changed comments to `subnode management`

File: AsyncDisplayKit/ASDisplayNode.h [hunk 2]
```diff
@@ -873,6 +891,17 @@ NS_ASSUME_NONNULL_BEGIN
...
 - (void)cancelLayoutTransitionsInProgress ASDISPLAYNODE_DEPRECATED;
 
+/**
+ * @abstract A boolean that shows whether the node automatically inserts and removes nodes based on the presence or
+ * absence of the node and its subnodes is completely determined in its layoutSpecThatFits: method.
+ *
+ * @discussion If flag is YES the node no longer require addSubnode: or removeFromSupernode method calls. The presence
+ * or absence of subnodes is completely determined in its layoutSpecThatFits: method.
+ *
+ * @deprecated Deprecated in version 2.0: Use automaticHierarchy
```
Comment (Adlai-Holler) [inline AsyncDisplayKit/ASDisplayNode.h]: One more comment needing update

File: AsyncDisplayKit/ASDisplayNode.h [hunk 3]
```diff
@@ -796,8 +796,24 @@ NS_ASSUME_NONNULL_BEGIN
 @end
 
 /*
- ASDisplayNode participates in ASAsyncTransactions, so you can determine when your subnodes are done rendering.
- See: -(void)asyncdisplaykit_asyncTransactionContainerStateDidChange in ASDisplayNodeSubclass.h
+ * ASDisplayNode support for automatic subnode management.
+ */
+@interface ASDisplayNode (AutomaticHierarchyManagement)
```
Comment (Adlai-Holler) [inline AsyncDisplayKit/ASDisplayNode.h]: One more category name needing update

Comment (maicki) [inline AsyncDisplayKit/ASDisplayNode.h]: giphy

Comment (Adlai-Holler) [inline AsyncDisplayKit/ASDisplayNode.h]: It's silly I know

Comment (maicki) [inline AsyncDisplayKit/ASDisplayNode.h]: Haha, no not at all! Just made fun of myself as I overlooked that. But should be updated now!

**Comment by Adlai-Holler** [comment]
Damn, could you give this a rebase when there's time @maicki ?

**Comment by maicki** [comment]
@Adlai-Holler Already did it :)

File: AsyncDisplayKit/Private/ASDisplayNode+UIViewBridge.mm [hunk 4]
```diff
@@ -344,7 +344,7 @@ - (void)setNeedsLayout
     // The node is loaded but we're not on main.
     // We will call [self __setNeedsLayout] when we apply
     // the pending state. We need to call it on main if the node is loaded
-    // to support implicit hierarchy management.
+    // to support automatic subnode management.
```
Comment (hannahmbanana) [inline AsyncDisplayKit/Private/ASDisplayNode+UIViewBridge.mm]: This was Adlai's comment from the closed PR: 
"Let's be extremely consistent here and use `to support automatic subnode management` and in the other comments."

**Comment by hannahmbanana** [comment]
@maicki: when you have time, could you fix the very minor nit above? Thanks!

**Pull request merged**